mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -282,6 +272,23 @@ def settings_path(self):
             return self._old_settings_path
         return join(xdg.BaseDirectory.save_config_path(get_xdg_base(), 'skills', self.skill_id), 'settings.json')
 
+    @property
+    def settings(self):
+        if self._settings is not None:
+            return self._settings
+        else:
+            LOG.error('Skill not fully initialized. Move code ' +
+                      'from  __init__() to initialize() to correct this.')
+            LOG.error(simple_trace(traceback.format_stack()))
+            raise Exception('Accessed self.settings in __init__')
+
+    @settings.setter
+    def settings(self, val):
```

Comment:
consider adding a Type annotation here since the property returns a different type than the setter

Reply:
JsonDatabase is a dict subclass, it is the same type just with extra methods for saving